### PR TITLE
Extend rules from a base "coding-standard" directory

### PR DESCRIPTION
**Pull request by BrianSipple**
As discussed in #61, (in particular, see [here](https://github.com/ember-cli/ember-cli-eslint/pull/61#issuecomment-218058715) and [here](https://github.com/ember-cli/ember-cli-eslint/pull/61#issuecomment-218246695)), this PR allows us to keep of our base configuration in its own directory, and then generates `.eslintrc.js` files with just a single setting:

```
extends: './node_modules/ember-cli-eslint/coding-standard/ember-application.js'
```
or
```
extends: '../node_modules/ember-cli-eslint/coding-standard/ember-testing.js'
```
Easy to understand what we’re doing. Easy to override if you don’t like it 😄.

I also updated `README.md` to outline our new approach.

One question I do still have is about versioning this change.  Part of me thinks that this setup will allow us to, occasionally, change settings in our base configs — and thus, since each of those would potentially constitute a breaking change, it might be good to set the stage now at 2.0. The other part of me, though, thinks that the other part of me is overthinking it, and would totally be okay with an incremental minor version bump .

cc @rwjblue @Turbo87 @mmun @alexlafroscia.

### CODE REVIEW

**Comment by Turbo87** [comment]
as mentioned in the other discussions I'm semi-happy about this, but if you agree that it's useful and a better approach then I'm fine with it.

File: README.md [hunk 1]
```diff
@@ -6,28 +6,13 @@
...
 ```
-npm install --save-dev babel-eslint@6.0.0-beta.6
+ember install ember-cli-eslint
 ```
 
-After installing, simply add the following option to your `eslint` configuration file:
-```
-    "parser": "babel-eslint",
-```
+This will create a `.eslintrc.js` file in the root of your project, and another `.eslintrc.js` file inside of `/test`. These files extend from our recommended configurations for [Ember application code](/best-practices/ember-application) and [Ember testing code](/best-practices/ember-test), respectively. However, starting from scratch is as easy as deleting the `extends` declaration and [writing your own configuration rules as usual](http://eslint.org/docs/user-guide/configuring).
```
Comment (alexlafroscia) [inline README.md]: How do these `/best-practices/...` links work?  What are they pointing at?

Comment (BrianSipple) [inline README.md]: Good catch -- these are actually attempting to reference the configs in `/coding-standard`, but I had originally named the directory `best-practices`. I'll update and make sure they lead to the right place.

Comment (alexlafroscia) [inline README.md]: Would it make more sense to make those full URLs instead?  That way they continue to work if someone is readying the `README` locally or from the [readme page](https://github.com/ember-cli/ember-cli-eslint/blob/master/README.md).

Comment (BrianSipple) [inline README.md]: I've always used relative links in markdown for files within the project -- but your question made me do some digging. It turns out [GitHub recommends always using relative URLs](https://help.github.com/articles/relative-links-in-readmes/).

File: blueprints/ember-cli-eslint/index.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
   name: 'ember-cli-eslint',
 
-  normalizeEntityName: function() {
+  normalizeEntityName() {
```
Comment (ro0gr) [inline blueprints/ember-cli-eslint/index.js]: Looks like it should not work under node@0.12?

File: index.js [hunk 3]
```diff
@@ -5,6 +5,9 @@ var jsStringEscape = require('js-string-escape');
 module.exports = {
   name: 'ember-cli-eslint',
 
+  // TODO: Disable this (or set it to return false) before committing
+  isDevelopingAddon: () => true,
```
Comment (morhook) [inline index.js]: nitpicking: You'll remember to take this one out before merging, right?

Comment (BrianSipple) [inline index.js]: That's no nitpick -- thanks for catch!

File: README.md (hunk 1 above)
Comment (alexlafroscia) [inline README.md]: Huh, interesting! I'm glad to know that Github actually has a recommendation, and what it is.

**Comment by Turbo87** [comment]
```
Cannot find module 'babel-eslint'
Referenced from: /Users/tbn/Code/skylines/ember/.eslintrc.js
Error: Cannot find module 'babel-eslint'
Referenced from: /Users/tbn/Code/skylines/ember/.eslintrc.js
    at Object.ModuleResolver.resolve (/Users/tbn/.nvm/versions/node/v5.11.0/lib/node_modules/eslint/lib/util/module-resolver.js:75:19)
    at load (/Users/tbn/.nvm/versions/node/v5.11.0/lib/node_modules/eslint/lib/config/config-file.js:519:42)
    at /Users/tbn/.nvm/versions/node/v5.11.0/lib/node_modules/eslint/lib/config/config-file.js:392:36
    at Array.reduceRight (native)
    at applyExtends (/Users/tbn/.nvm/versions/node/v5.11.0/lib/node_modules/eslint/lib/config/config-file.js:363:28)
    at Object.load (/Users/tbn/.nvm/versions/node/v5.11.0/lib/node_modules/eslint/lib/config/config-file.js:531:22)
    at loadConfig (/Users/tbn/.nvm/versions/node/v5.11.0/lib/node_modules/eslint/lib/config.js:64:33)
    at getLocalConfig (/Users/tbn/.nvm/versions/node/v5.11.0/lib/node_modules/eslint/lib/config.js:126:23)
    at Config.getConfig (/Users/tbn/.nvm/versions/node/v5.11.0/lib/node_modules/eslint/lib/config.js:227:22)
    at processText (/Users/tbn/.nvm/versions/node/v5.11.0/lib/node_modules/eslint/lib/cli-engine.js:223:27)
```

when running `eslint app` from `/Users/tbn/Code/skylines/ember/`

can we work around that?

**Comment by Turbo87** [comment]
actually this prevents us from using the `--fix` option, which is most certainly undesirable...

**Comment by rwjblue** [comment]
Weird. Why does it not work?

File: index.js [hunk 4]
```diff
@@ -5,6 +5,11 @@ var jsStringEscape = require('js-string-escape');
 module.exports = {
   name: 'ember-cli-eslint',
 
+  // TODO: Disable this (or set it to return false) before committing
+  isDevelopingAddon: function() {
```
Comment (rwjblue) [inline index.js]: O_o

**Comment by Turbo87** [comment]
> Why does it not work?

the new default config depends on `babel-eslint`, but if that is installed to `node_modules/ember-cli-eslint/node_modules/babel-eslint` the `eslint` CLI isn't able to find it like this.

**Comment by rwjblue** [comment]
I see. Perhaps this is a bug in eslint. 

The parser should either be resolved based on the config location that specifies the parser option, or we should be allowed to pass a non-string (so that ember-cli-eslint can require it in the extended config file).

**Pull request merged**